### ISSUE TITLE
Java Android Library Support

### ISSUE THREAD

**Issue by realbarisbasturk** [Kotlin/binary-compatibility-validator#94]
Binary compatibility plugin configures necessary tasks like 'apiDump' and 'apiCheck' only if `kotlin`, `kotlin-android` or `kotlin-multiplatform` plugins are applied as can be seen [here](https://github.com/Kotlin/binary-compatibility-validator/blob/89ad35ec75dcbb6ede9b72f7185416783828db02/src/main/kotlin/BinaryCompatibilityValidatorPlugin.kt#L112) 

Although it can be used for Java libraries, due to missing the plugins listed above, necessary tasks are not created.

Java Android libraries can be supported if the tasks are also created for projects where `com.android.library` plugin is applied.

**Comment by qwwdfsad** [comment]
Unfortunately, `kotlin-android` and `com.android.library` interact in an unpredictable way that prevents major libs from updating (#103, as well as few other libs I've tested).

Unfortunately, I'm short on time in order to properly investigate and fix it, so in order to unblock libraries from updating, I'm reverting this feature and releasing 0.12.1

**Comment by realbarisbasturk** [comment]
I did not test the case where plugin is applied to the root project (as it is recommended). Since the functional tests I added also does not cover this case but instead applies the plugin to the individual java/kotlin modules, I missed it during development. Thanks for sharing your findings, I will look into it asap (hopefully this weekend) and create a new PR to fix the issue.

**Comment by realbarisbasturk** [comment]
Hi @qwwdfsad, it's been a while since I've created a PR for this issue. I'd appreciate if you could take a look when you have a chance.

**Comment by daymxn** [comment]
Any update on this? I see that #108 was closed, but couldn't see any reason why?

cc @qwwdfsad @realbarisbasturk

**Comment by fzhinkin** [comment]
@daymxn I don't know the reason behind #108 being closed, however better Android libraries support is being considered as part of #223. No strong commitments, though.
